src/shared/components/query/CancerStudySelector.tsx

```diff
@@ -134,6 +134,20 @@ export default class CancerStudySelector extends React.Component<ICancerStudySel
         );
     });
 
+    @computed
+    get checkReferenceGenome() {
+        let genomeValid = true;
+        if (this.store.selectableSelectedStudies.length > 0) {
+            const referenceGenome = this.store.selectableSelectedStudies[0].referenceGenome;
```

Comment:
i think you could map to an array of all referneceGenomes, then call _.uniq on that array, then measure length > 1 to determine validity of query

Reply:
also, do we allow user to submit query when checkReferenceGenome is false (mixed genomes) ? Or do we block them from doing so, as we do if they select too many studies?

Reply:
we do not allow merge studies from different genomes.